apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,317 @@
+}
+
+static wifi_manager_result_e wm_connect_by_rssi_init(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_init(&g_wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to init wifi_manager");
+		return res;
+	}
+	WT_LEAVE;
+	return res;
+}
+
+static wifi_manager_result_e wm_connect_by_rssi_deinit(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "WiFi Manager failed to stop");
+		return res;
+	}
+	WT_LEAVE;
+	return res;
+}
+
+static wifi_manager_result_e _wt_scan(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_scan_ap(NULL);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to scan");
+		return res;
+	}
+	WM_TEST_WAIT; // wait the scan result
+	WT_LEAVE;
+	return res;
+}
+
+static wifi_manager_result_e _wt_connect(void *arg)
+{
+	WT_ENTER;
+	struct wt_options *ap_info = (struct wt_options *)arg;
+	wifi_manager_ap_config_s apconfig;
+	strncpy(apconfig.ssid, ap_info->ssid, WIFIMGR_SSID_LEN);
+	apconfig.ssid_length = strlen(ap_info->ssid);
+	if (apconfig.ssid_length > WIFIMGR_SSID_LEN) {
+		apconfig.ssid[WIFIMGR_SSID_LEN + 1] = '\0';
+	} else {
+		apconfig.ssid[apconfig.ssid_length + 1] = '\0';
+	}
+	apconfig.ap_auth_type = ap_info->auth_type;
+	if (ap_info->auth_type != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(apconfig.passphrase, ap_info->password, WIFIMGR_PASSPHRASE_LEN);
+		apconfig.passphrase_length = strlen(ap_info->password);
+		if (apconfig.passphrase_length > WIFIMGR_PASSPHRASE_LEN) {
+			apconfig.passphrase[WIFIMGR_PASSPHRASE_LEN + 1] = '\0';
+		} else {
+			apconfig.passphrase[apconfig.passphrase_length + 1] = '\0';
+		}
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	} else {
+		apconfig.passphrase[0] = '\0';
+		apconfig.passphrase_length = 0;
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	}
+	wt_print_wifi_ap_profile(&apconfig, "Connecting AP Info");
+	wifi_manager_result_e res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "AP connect failed");
+		return res;
+	}
+	/* Wait for DHCP connection */
+	WT_LOG(TAG, "wait join done");
+	WM_TEST_WAIT;
+	WT_LEAVE;
+	return res;
+}
+
+static void _wt_disconnect()
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	/* Disconnect AP */
+	res = wifi_manager_disconnect_ap();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to disconnect ap (%d)", res);
+		return;
+	}
+	WM_TEST_WAIT;
+	WT_LEAVE;
+}
+
+
+static void wm_run_connect_by_rssi(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e ret = WIFI_MANAGER_SUCCESS;
+	g_opt = (struct wt_options *)arg;
+	int passed = 0, failed = 0;
+	for (int i = 0; i < g_opt->repeat; i++) {
+		WT_LOG(TAG, "====== Iteration %d starts ======", i + 1);
+		ret = wm_connect_by_rssi_init();
+		if (ret != WIFI_MANAGER_SUCCESS) {
+			WT_LOGE(TAG, "wm_connect_by_rssi_init failed");
```

Comment:
at every point when an API fails, retry is done with a delay, 
please check is delay required here also.